components/sock_utils/src/gai_strerror.c

```diff
@@ -0,0 +1,16 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdio.h>
+#include "gai_strerror.h"
+
+const char *gai_strerror(int ecode)
+{
+    static char str[32];
```

Comment:
Returning a static buffer is not thread safe.

Reply:
need to create a hardcoded list of errors.

Reply:
consider the TLS here `__thread_local_static...`

Reply:
used TLS